dev-tools/omdb/src/bin/omdb/db/whatis.rs

```diff
@@ -0,0 +1,158 @@
+    // system.
+    let query = "
+        WITH constraint_column_counts AS (
+            SELECT
+                constraint_catalog,
+                constraint_schema,
+                constraint_name,
+                table_catalog,
+                table_schema,
+                table_name,
+                COUNT(*) as column_count
+            FROM
+                information_schema.key_column_usage
+            GROUP BY
+                constraint_catalog,
+                constraint_schema,
+                constraint_name,
+                table_catalog,
+                table_schema,
+                table_name
+        )
+        SELECT DISTINCT
+            tc.table_name,
+            kcu.column_name
+        FROM
+            information_schema.table_constraints tc
+            JOIN information_schema.key_column_usage kcu
+                ON tc.constraint_catalog = kcu.constraint_catalog
+                AND tc.constraint_schema = kcu.constraint_schema
+                AND tc.constraint_name = kcu.constraint_name
+                AND tc.table_catalog = kcu.table_catalog
+                AND tc.table_schema = kcu.table_schema
+                AND tc.table_name = kcu.table_name
+            JOIN information_schema.columns c
+                ON kcu.table_catalog = c.table_catalog
+                AND kcu.table_schema = c.table_schema
+                AND kcu.table_name = c.table_name
+                AND kcu.column_name = c.column_name
+            JOIN constraint_column_counts ccc
+                ON tc.constraint_catalog = ccc.constraint_catalog
+                AND tc.constraint_schema = ccc.constraint_schema
+                AND tc.constraint_name = ccc.constraint_name
+                AND tc.table_catalog = ccc.table_catalog
+                AND tc.table_schema = ccc.table_schema
+                AND tc.table_name = ccc.table_name
+            LEFT JOIN pg_indexes idx
+                ON idx.tablename = tc.table_name
+                AND idx.indexname = tc.constraint_name
+                AND idx.schemaname = tc.table_schema
+        WHERE
+            tc.constraint_type IN ('PRIMARY KEY', 'UNIQUE')
+            AND tc.table_schema = 'public'
+            AND c.data_type = 'uuid'
+            AND ccc.column_count = 1
+            AND (idx.indexdef IS NULL OR idx.indexdef NOT LIKE '% WHERE %')
+        ORDER BY
+            tc.table_name, kcu.column_name
+    ";
+
+    #[derive(Debug, diesel::QueryableByName)]
+    struct UniqueUuidColumn {
+        #[diesel(sql_type = diesel::sql_types::Text)]
+        table_name: String,
+        #[diesel(sql_type = diesel::sql_types::Text)]
+        column_name: String,
+    }
+
+    let unique_columns: Vec<UniqueUuidColumn> = diesel::sql_query(query)
+        .load_async(&*conn)
+        .await
+        .context("querying information_schema for unique UUID columns")?;
+
+    // Search separately for each UUID provided on the command line.
+    for uuid in &args.uuids {
```

Comment:
niitpick, considered pedantic: if you put a 
```suggestion
    for &uuid in &args.uuids {
```
here, `uuid` will already be dereferenced by the `for` loop and you don't have to write `*uuid` several times in the subsequent code.